frontmatter_parsers.ts

```diff
@@ -0,0 +1,39 @@
+import { titleCase } from "./deps/case.ts"
+import * as $ from "./deps/scale.ts"
+import * as datetime from "./deps/std/datetime.ts"
+import { FrontmatterParser } from "./frontmatter.ts"
+import { splitFirst } from "./util/splitFirst.ts"
+
+export const title: FrontmatterParser<string> = (raw) => {
+  $.assert($.str, raw)
+  return titleCase(raw)
+}
+
+export const description: FrontmatterParser<string> = (raw) => {
+  $.assert($.str, raw)
+  return raw
+}
+
+const $stability = $.literalUnion(["experiment", "unstable", "nearing", "stable"])
```

Comment:
i tried adding stability to the example but it doesnt seem to like it.

```ts
/**
 * @title Example
 * @tags example,docs,deno,typescript
 * @date 10-10-2023
 * @description This description can be very long and even span
 * @stability experiment
 * multiple lines. It will be parsed as expected.
 */

// Welcome to egts. This very script is an egts.
// The jsdoc section above serves as frontmatter.
// The code serves as to-be markdown code fences.
// And the comments serve as ordinary lines of markdown body.

// Bring imports into scope.
import { compile } from "egts"
import { date, description, stability, tags, title } from "egts_frontmatter_parsers"

// Compile an egts of this very script!
const result = compile("basic.eg.ts", await Deno.readTextFile(new URL(import.meta.url)), {
  title,
  tags: tags(["example", "docs", "deno", "typescript"]),
  date: date("dd-MM-yyyy"),
  description,
  stability,
})
```

```ts
➜  trun git:(reworking) ✗ deno run -A examples/basic.eg.ts
pieces [ "experiment\n", "multiple lines. It will be parsed as expected." ]
error: Uncaught (in promise) Error: Failed to parse "stability" from "basic.eg.ts"
ScaleAssertError: value invalid value
    at Codec._assert (https://deno.land/x/scale@v0.11.2/codecs/union.ts:89:15)
    at Module.assert (https://deno.land/x/scale@v0.11.2/common/codec.ts:140:9)
    at stability (file:///home/ryan/Documents/Repositories/trun/frontmatter_parsers.ts:22:17)
    at parse (file:///home/ryan/Documents/Repositories/trun/frontmatter.ts:36:37)
    at compile (file:///home/ryan/Documents/Repositories/trun/mod.ts:9:33)
    at file:///home/ryan/Documents/Repositories/trun/examples/basic.eg.ts:20:16
      throw new Error(`Failed to parse "${key}" from "${pathname}"\n${Deno.inspect(e)}`)
            ^
    at parse (file:///home/ryan/Documents/Repositories/trun/frontmatter.ts:38:13)
    at compile (file:///home/ryan/Documents/Repositories/trun/mod.ts:9:33)
    at file:///home/ryan/Documents/Repositories/trun/examples/basic.eg.ts:20:16
➜  trun git:(reworking) ✗ 
```
```

Reply:
Very interesting. Seems it's including the `\n` in the value that it's asserting. Aka., this parser should trim `raw` before asserting.

Reply:
Should be fixed now. Wanna try again?

Reply:
```
export const stability: FrontmatterParser<string> = (raw) => {
  $.assert($.str, raw)
  const pieces = splitFirst(raw.split("\n").join(" "), " ")
  if (pieces) $.assert($stability, pieces[0])
  else $.assert($stability, raw)
  return raw
}

```

Reply:
Decided just to get rid of the stability-associated descriptions parsing. Only applies to unstable/experiment anyways. We can enable this with an `@unstable` or `@experiment`, which may be better (as opposed to forcing every example to have a `@stability`).

Reply:
sounds good